test/e2e/operator/garden/internal/rotation/observability.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
Is it possible to make https://github.com/gardener/gardener/blob/master/test/e2e/gardener/shoot/internal/rotation/observability.go reusable instead of duplicating the content? I have tried to do so for a few other similar code lines, e.g. https://github.com/gardener/gardener/blob/master/test/utils/rotation/etcd_encryption_key.go. Can we achieve the same here?

Reply:
Keeping it separate is better option, as there are quite a few small difference between two.

Reply:
Please take a look at https://github.com/rfranzke/gardener/commit/00d649804e88825a4a4d6b43c2837368dcd468aa

Reply:
I didn't want to pass a function for each thing. But this looks better Thanks. I have cherry-picked - https://github.com/rfranzke/gardener/commit/00d649804e88825a4a4d6b43c2837368dcd468aa in this PR.